packages/edit-site/src/components/sidebar/global-styles-v2/screens/typography-element.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	ButtonGroup,
+	CardBody,
+	Dropdown,
+	DropdownMenu,
+	DropdownMenuItem,
+	DropdownTrigger,
+	FormGroup,
+	Grid,
+	ListGroup,
+	Select,
+	Slider,
+	TextInput,
+	UnitInput,
+	useNavigatorParams,
+	VStack,
+} from '@wordpress/components';
+import {
+	formatCapitalize,
+	formatLowercase,
+	formatUppercase,
+} from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { Screen, ScreenHeader } from '../components';
+import { useAppState } from '../state';
+
+const typographyOptions = [
+	{
+		label: 'Font',
+		value: 'fontFamily',
+		defaultValue: 'Helvetica Neue',
+	},
+	{
+		label: 'Size',
+		value: 'fontSize',
+		defaultValue: '14px',
+	},
+	{
+		label: 'Appearance',
+		value: 'fontWeight',
+		defaultValue: '400',
+	},
+	{
+		label: 'Line height',
+		value: 'lineHeight',
+		defaultValue: '1.5',
+	},
+	{
+		label: 'Letter spacing',
+		value: 'letterSpacing',
+		defaultValue: '0',
+	},
+	{
+		label: 'Letter case',
+		value: 'letterCase',
+		defaultValue: null,
+	},
+];
+
+const TypographyOptions = () => {
+	const {
+		attributes: { styles },
+		toggleAttribute,
+	} = useStyleAttributesForScreen();
+
+	const options = typographyOptions.map( ( o ) => {
+		return {
+			...o,
+			isSelected: ! isUndefined( styles[ o.value ] ),
+			onClick: () => toggleAttribute( o.value, o.defaultValue ),
+		};
+	} );
+
+	return (
+		<Dropdown>
+			<DropdownTrigger icon="ellipsis" isControl isSubtle size="small" />
+			<DropdownMenu>
+				{ options.map( ( option ) => (
+					<DropdownMenuItem key={ option.value } { ...option }>
```

Comment:
In the original PR we're setting `key` manually above, but ESLint won't understand we're doing it, so we're specifically providing it as a `key` here, along with the rest of `option`.